### ISSUE TITLE
Add support for `#[track_caller]`

### ISSUE THREAD

**Issue by shepmaster** [shepmaster/snafu#247]
This will involve adding a feature flag for Rust 1.46, then adding `#[track_caller]` to the context selector methods like `build` and `fail`.

**Comment by ctaggart** [comment]
I found the RFC at https://github.com/rust-lang/rfcs/pull/2091. I think this is what I'm looking for. Yesterday, I ended up sprinkling [code like this](https://github.com/ctaggart/autorust/pull/74/files#diff-9b76d076a80f534dc21ae3b4c28b82c56690f7d009025261b35a6abd57a2d254R222-R225) everywhere:
``` rust
        let typ = ident(&alias_name.to_camel_case()).context(IdentError {
            file: file!(),
            line: line!(),
        })?;
```
It would be great if there was a variable like `source` that captured the `caller` automatically.

**Comment by shepmaster** [comment]
> captured the `caller` automatically.

I have doubts about the true value of this. Why do you reuse an error type so frequently? Concretely speaking, why do you use `IdentError` in so many places that the file / line location is required to be able to make sense of a resulting error? If each context selector is used once, its name is already a unique identifier to locate the exact location of the problem.

**Comment by ctaggart** [comment]
Thanks @shepmaster. I'd like to get to a point where each one is uniquely named, but it was easiest at first to wrap an error type generically. Well, until the error was thrown and I couldn't figure out the location.

**Comment by shepmaster** [comment]
Closed via #247
